### PR TITLE
Use mkdirp and rimraf instead of their unix equivalents to ensure cro…

### PR DESCRIPTION
**Pull request by nerdbeere**
…ss platform compatability

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
     "watch:style": "nodemon --watch style --ext scss --no-stdin --exec \"npm run build:style\"",
-    "copy-files": "rm -rf build/fonts && mkdir -p build/fonts && cp ./node_modules/font-awesome/fonts/* build/fonts/ && cp -r ./resource/* ./build/",
-    "package": "npm run build && rm -rf ./package &&  mkdir -p ./package && mkdir -p ./binaries && browserify --node -e build/index.js -o ./package/index.js --exclude remote && npm run -s copy-files-package && npm run -s create-binaries",
+    "copy-files": "rimraf build/fonts && mkdirp build/fonts && cp ./node_modules/font-awesome/fonts/* build/fonts/ && cp -r ./resource/* ./build/",
+    "package": "npm run build && rimraf ./package &&  mkdirp ./package && mkdirp ./binaries && browserify --node -e build/index.js -o ./package/index.js --exclude remote && npm run -s copy-files-package && npm run -s create-binaries",
     "create-binaries": "electron-packager ./package James --all --overwrite --icon=./resource/icon.icns --version=0.28.3 --out binaries",
     "zip-binaries": "cd binaries; rm *.zip; for i in *; do zip --symlinks -r \"${i%/}.zip\" \"$i\"; done",
```
Comment (davidneat) [inline package.json]: zip-binaries still uses the rm command. please use rimraf instead.

Comment (nerdbeere) [inline package.json]: this command won't work anyway because of `zip`

Comment (davidneat) [inline package.json]: Maybe https://www.npmjs.com/package/node-zip will do the job

Comment (nerdbeere) [inline package.json]: It's not only the zip command but the whole scripting in there needs to be rewritten with node

**Comment by nerdbeere** [comment]
@mitchhentges how did you solve the issues with npm scripts when running them on windows? Is this PR interesting for you?

**Comment by mitchhentges** [comment]
Hah, I opened the project in Linux instead. I can definitely look at this
pr in a couple of days though

**Comment by mitchhentges** [comment]
Looks like the single quotes don't work on windows:

´´´
C:\Users\MDF\Documents\dev\james>npm run build

> james@1.2.3 build C:\Users\MDF\Documents\dev\james
> rimraf build && parallelshell 'npm run build:js' 'npm run build:sty

''npm' is not recognized as an internal or external command,
operable program or batch file.
'run' is not recognized as an internal or external command,
operable program or batch file.
The filename, directory name, or volume label syntax is incorrect.
''npm' is not recognized as an internal or external command,
operable program or batch file.
'run' is not recognized as an internal or external command,
operable program or batch file.
The filename, directory name, or volume label syntax is incorrect.
´´´


I'm pretty sure that it is due to the single quotes in ´package.json´.
```
"rimraf build && parallelshell 'npm run build:js' 'npm run build:style'"
                               ^                ^ ^                   ^
```

Changing them to escaped double quotes should work according to https://github.com/keithamus/parallelshell/issues/1

```
"rimraf build && parallelshell \"npm run build:js\" \"npm run build:style\""
```

**Comment by nerdbeere** [comment]
I made that change but can't test it on a windows machine

**Comment by mitchhentges** [comment]
Now Windows is a little mad about `cp`. I've made a patch that works on my Windows machine, I'd recommend testing it on yours:
[patch.txt](https://github.com/uxebu/james/files/97529/patch.txt)

**Comment by mitchhentges** [comment]
Err, typo when adding the dev-dependency. The patch, for realsies:
[james.patch](https://github.com/uxebu/james/files/97532/patch.txt)

**Comment by nerdbeere** [comment]
I applied your patch, it works on my machine (OSX).

**Comment by mitchhentges** [comment]
Are we OK to merge?

**Pull request merged**